### PR TITLE
feat(*): support wrapper classes for Java basic types

### PR DESCRIPTION
**Pull request by Mulavar**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: array.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+	SetCollectionSerialize(&DoubleArray{})
+	SetCollectionSerialize(&CharacterArray{})
+}
+
+//BooleanArray Boolean[]
+type BooleanArray struct {
+	Values []bool
+}
+
+func (ba *BooleanArray) Get() []interface{} {
```
Comment (AlexStocks) [inline array.go:36]: Pls add comment "// nolint"

**Review by AlexStocks — APPROVED** [review]

File: array_test.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	"testing"
+)
+
+func TestBooleanArray(t *testing.T) {
+	booleanArray := &BooleanArray{[]bool{true, false}}
+	jcs := JavaCollectionSerializer{}
+	e := &Encoder{}
+	a := assert.New(t)
+
+	err := jcs.EncObject(e, booleanArray)
```
Comment (wongoo) [inline array_test.go:31]: not use the internal api to test, use `e.Encode(booleanArray)` instead.

**Review by wongoo — CHANGES_REQUESTED** [review]
@Mulavar use Encoder to test encode in unit tests please.

File: array.go [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+}
+
+// nolint
+func (*IntegerArray) JavaClassName() string {
+	return "[java.lang.Integer"
+}
+
+// ByteArray Byte[]
+type ByteArray struct {
+	IntegerArray
```
Comment (wongoo) [inline array.go:93]: why not use uint8 as byte value？

Comment (Mulavar) [reply array.go:93]: In the beginning, it was for simplicity, but you are right, and I will alse use []int16 as ShortArray.Values in the next commit.

Comment (Mulavar) [reply array.go]: done

**Comment by wongoo** [comment]
@Mulavar thanks for your effort. wrapper classes is mainly for java, so the unit test should contains cross language encode/decode.  Pls pay more time to do that, thanks very much!


eg:

**go encode -> java decode**: 
- go encode: https://github.com/apache/dubbo-go-hessian2/blob/e0c085c9ea5936e771ecc351c94609d5a56a7a1b/output/output.go#L44
- java decode: https://github.com/apache/dubbo-go-hessian2/blob/e0c085c9ea5936e771ecc351c94609d5a56a7a1b/test_hessian/src/test/java/unit/GoUserListTest.java#L33

**java encode-> go decode**:
- java encode: https://github.com/apache/dubbo-go-hessian2/blob/e0c085c9ea5936e771ecc351c94609d5a56a7a1b/test_hessian/src/main/java/test/TestCustomReply.java#L605
- go decode: https://github.com/apache/dubbo-go-hessian2/blob/e0c085c9ea5936e771ecc351c94609d5a56a7a1b/object_test.go#L791

**Review by wongoo — APPROVED** [review]

**Pull request merged**